packages/data-context/src/codegen/code-generator.ts

```diff
@@ -36,6 +36,9 @@ export async function codeGenerator (
   const templateFiles = await allFilesInDir(action.templateDir)
   const codeGenResults: CodeGenResults = { files: [], failed: [] }
 
+  // checks if the target directory exists and creates it if necessary
+  await fs.ensureDir(action.target)
```

Comment:
This is unnecessary, since the generator uses [fs.outputFile](https://github.com/jprichardson/node-fs-extra/blob/master/docs/outputFile.md#outputfilefile-data-options-callback) to generate files which will create the directory if it does not exist.

Reply:
Added it to replace this function `maybeCreateE2EDir()` that failed. 
Asking stats on a non-existing directory does not work.

I hope it is not useful.